### PR TITLE
[Testing] Execute functional tests for each filtering configuration

### PR DESCRIPTION
**Pull request by timgrein**
## Related to https://github.com/elastic/enterprise-search-team/issues/3736 ##

This PR adds the capability of automatically executing a functional test for each `filtering*.json` file for the corresponding source. This enables us to test filtering and also to do performance measurements for different filtering configurations.

Also clean up `mysql/fixture.py` by improving naming and logging.

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/kibana.py [hunk 1]
```diff
@@ -254,17 +260,17 @@ def _parser():
...
 
     return parser
 
 
 def _load_filtering(filtering_file_path):
-    if not os.path.exists(filtering_file_path):
+    if not filtering_file_path or not os.path.exists(filtering_file_path):
         logger.warn(
-            f"filtering file at '{filtering_file_path}' does not exist. Fallback to default filtering."
+            f"filtering file {('at' + filtering_file_path) if filtering_file_path else ''} does not exist. Fallback to default filtering."
```
Comment (artem-shelkovnikov) [inline connectors/kibana.py:273]: I'd move it to variable for readability.

File: connectors/kibana.py [hunk 2]
```diff
@@ -90,6 +90,12 @@
 }
 
 
+def nullable_str(string):
```
Comment (artem-shelkovnikov) [inline connectors/kibana.py:93]: Maybe just `nullable`? Cause I don't see that it does anything string-specific

File: connectors/sources/tests/fixtures/mysql/fixture.py [hunk 3]
```diff
@@ -22,44 +38,88 @@ def random_text(k=1024 * 20):
...
-        f"INSERT INTO customers_{table}"
-        + "(name, age, description) VALUES (%s, %s, %s)"
+class Customer:
+    def __init__(self, id_):
+        self.id_ = id_
+        self.name = fake.name()
+        self.description = BIG_TEXT
+
+    def to_row(self):
+        return self.id_, self.name, self.description
```
Comment (artem-shelkovnikov) [inline connectors/sources/tests/fixtures/mysql/fixture.py:48]: Honestly the class feels rudimentary - I did not look too deep into it, but it looks like you just replace it with a function similar to:

```
def customer_row(id):
    return id, fake.name(), BIG_TEXT
```

File: connectors/tests/ftest.sh [hunk 4]
```diff
@@ -37,34 +37,54 @@ fi
...
+
+  $PYTHON $ROOT_DIR/scripts/verify.py --index-name search-$NAME --service-type $NAME --size $SIZE
+}
+
+# run test once without filtering
+run_test
+
+for filtering in ./filtering*.json
+do
+  run_test $filtering
```
Comment (artem-shelkovnikov) [inline connectors/tests/ftest.sh:86]: Somehow it feels that `verify` can be different per filtering setup too - different number of entries should be present or some other criteria checked. Are you planning to add it in upcoming PRs?

**Review by artem-shelkovnikov — COMMENTED** [review]
Overall looks good - curious to hear @tarekziade's opinion too :)

File: connectors/tests/ftest.sh (hunk 4 above)
Comment (timgrein) [reply connectors/tests/ftest.sh:86]: I've created an issue an hour ago about addressing exactly this issue (and some others). I would've probably the time before 8.8 kick-off implementing the proposal...I'll double check, if there's room this week to address those. Feel free to comment on it.

File: connectors/sources/tests/fixtures/mysql/fixture.py (hunk 3 above)
Comment (timgrein) [reply connectors/sources/tests/fixtures/mysql/fixture.py:48]: Fair point, thought about externalising a more complex `Customer` class, which can be reused in different tests. But for this PR it's probably better to just us a one-liner function.

Comment (artem-shelkovnikov) [reply connectors/sources/tests/fixtures/mysql/fixture.py:48]: TBH I think that functional tests should be as simple as possible. If they get complex, it's a good question - do we really want that complexity?

**Comment by timgrein** [comment]
Closing this as we agreed to start a bigger refactoring around functional tests, which will make this change obsolete.

**Pull request closed without merging**